src/pac.c

```diff
@@ -435,6 +435,43 @@ pac_stash_decayed(tsdn_t *tsdn, pac_t *pac, ecache_t *ecache,
 	return nstashed;
 }
 
+static bool
+decay_with_process_madvise(edata_list_inactive_t *decay_extents) {
+	cassert(have_process_madvise);
+	assert(opt_process_madvise_max_batch > 0);
+#ifndef JEMALLOC_HAVE_PROCESS_MADVISE
+	return true;
+#else
+	size_t stack_limit = PROCESS_MADVISE_MAX_BATCH_LIMIT;
+	size_t len = stack_limit < opt_process_madvise_max_batch ? stack_limit :
```

Comment:
With the clip we should be safe to assert `opt_process_madvise_max_batch <= PROCESS_MADVISE_MAX_BATCH_LIMIT` instead of still using `len` here?

Reply:
The only case not covered by the config `clip` is the default value `PROCESS_MADVISE_MAX_BATCH_DEFAULT`. To be extra safe, added a manual clip in init_process_madvise.